stdlib/SparseArrays/src/linalg.jl

```diff
@@ -376,6 +376,50 @@ function dot(x::SparseVector, A::AbstractSparseMatrixCSC, y::SparseVector)
     r
 end
 
+function dot(A::Matrix{TA}, B::SparseMatrixCSC{TB}) where {TA, TB}
```

Comment:
```suggestion
function dot(A::AbstractMatrix{TA}, B::SparseMatrixCSC{TB}) where {TA, TB}
```
? Otherwise this looks very specific. It might trigger some ambiguities, but then you might wish to consider also addressing them. Well, or at least it would be interesting to see which ambiguities pop up.

Reply:
Is it a good default? This means all new matrix types `M` will need to implement `dot(::M, ::SparseMatrixSCS)` in order to have an implementation that prevails, otherwise all calls to `dot(::M, ::SparseMatrixSCS)` will be ambiguous, that's potentially a lot

Reply:
Hm, I see. That raises even more the question what to do with structured "sparse" matrix types.

Reply:
some other people could voice in, but this is why I had the idea of widening the type to all the dense matrices from LinearAlgebra